robot_smach_states/src/robot_smach_states/navigation/follow_operator2_0.py

```diff
@@ -0,0 +1,475 @@
+        self._robot = robot
+        self._operator_name = "operator"
+        self._lost_timeout = lost_timeout
+        self._lost_distance = lost_distance
+        self._face_pos_pub = rospy.Publisher('/%s/follow_operator/operator_detected_face' % robot.robot_name,
+                                             geometry_msgs.msg.PointStamped, queue_size=10)
+
+    def execute(self, userdata=None):
+        rospy.loginfo("Trying to recover the operator")
+        self._robot.head.look_at_standing_person()
+        self._robot.speech.speak("%s, please look at me while I am looking for you" % self._operator_name, block=False)
+
+        # Wait for the operator and find his/her face
+        operator_recovery_timeout = self._lost_timeout
+        start_time = rospy.Time.now()
+
+        look_distance = 2.0  # magic number 4
+        look_angles = [0.0,  # magic numbers
+                       math.pi / 6,
+                       math.pi / 4,
+                       math.pi / 2.3,
+                       0.0,
+                       -math.pi / 6,
+                       -math.pi / 4,
+                       -math.pi / 2.3]
+        head_goals = [kdl_conversions.VectorStamped(x=look_distance * math.cos(angle),
+                                                    y=look_distance * math.sin(angle), z=1.7,
+                                                    frame_id="/%s/base_link" % self._robot.robot_name)
+                      for angle in look_angles]
+
+        i = 0
+        while (rospy.Time.now() - start_time).to_sec() < operator_recovery_timeout:
+            if self.preempt_requested():
+                return False
+
+            self._robot.head.look_at_point(head_goals[i])
+            i += 1
+            if i == len(head_goals):
+                i = 0
+            self._robot.head.wait_for_motion_done()
+            raw_detections = self._robot.perception.detect_faces()
+            best_detection = self._robot.perception.get_best_face_recognition(raw_detections, "operator")
+
+            rospy.loginfo("best_detection = {}".format(best_detection))
+            if best_detection:
+                roi = best_detection.roi
+                try:
+                    operator_pos_kdl = self._robot.perception.project_roi(roi=roi, frame_id="map")
+                except Exception as e:
+                    rospy.logerr("head.project_roi failed: %s", e)
+                    return 'Failed'
+                operator_pos_ros = kdl_conversions.kdlVectorStampedToPointStamped(operator_pos_kdl)
+                self._face_pos_pub.publish(operator_pos_ros)
+
+                recovered_operator = self._robot.ed.get_closest_laser_entity(radius=self._lost_distance,
+                                                                             center_point=operator_pos_kdl)
+                if recovered_operator:
+                    print "Found one!"
+                    self._operator_id = recovered_operator.id
```

Comment:
`self._operator_id` doesn't do anything except that it's used for printing. Why set it as a member attribute?